### ISSUE TITLE
tabindex="-1" on last element prevents Tab key from cycling back to the top

### ISSUE THREAD

**Issue by Sammii** [theKashey/react-focus-lock#40]
Scenario:
The last focusable element in the modal has attribute `tabindex="-1"` or CSS `visibility: hidden` and is such not tabbable by the browser.

Expected Result:
When using Tab key to navigate, this element is skipped, so the expectation is for focus to cycle back to the first tabbable element at the top of the modal.

Actual Result:
When using Tab key to navigate, the focus remains stuck on the last tabbable element in the modal and does not cycle back to the top.

**Comment by theKashey** [comment]
I'll check. Something is very wrong, as long I could not find any reason for this strange behavior.

**Comment by Sammii** [comment]
Thanks very much @theKashey ! I have some further clarification of the error scenario that we are encountering:

If the last tabbable link in the modal is within a parent that has `display: none` *and* the link does not have `tabindex="-1"` explicitly specified, then the focus cycle gets stuck, although browser does not tab to this link (since it is hidden).

Example:

Scenario 1: ❌  Focus stuck at `#link1`
```
<FocusLock>
  ...
  <a id="link1">Test</a>
  <div style="display: none;">
    <a id="link2">Test</a>
  </div>
</FocusLock>
```

Scenario 2: ✅  Cycles back to top as expected
```
<FocusLock>
  ...
  <a id="link1">Test</a>
  <div style="display: none;">
    <a id="link2" tabindex="-1">Test</a>
  </div>
</FocusLock>
```

Scenario 3: ❌  Focus stuck at `#link1`
```
<FocusLock>
  ...
  <a id="link1">Test</a>
  <div style="display: none;">
    <a id="link2">Test</a>
  </div>
  <a id="link3" tabindex="-1">Test</a> /* This cannot be tabbed to */
</FocusLock>
```

Scenario 4:  ✅  Cycles back to top as expected
```
<FocusLock>
  ...
  <a id="link1">Test</a>
  <div style="display: none;">
    <a id="link2">Test</a>
  </div>
  <a id="link3">Test</a>
</FocusLock>
```

**Comment by theKashey** [comment]
Yep, I've found the problem already - underlaying logic _removes_ `tabindex=-1` elements from two lists - "my" focusable elements, and the "rest" focusable elements.
Meanwhile it also has a logic about removing invisible (you forgot `input type=hidden`) elements, but it work only one list, not for another.
Result - 💣 💥

**Comment by Sammii** [comment]
🙏  Thank you @theKashey ! Your patch works like a charm!

**Comment by theKashey** [comment]
Thank you in response. You helped to find and close quite important issue.
